ens/main.py

```diff
@@ -98,15 +99,18 @@ def reverse_domain(address: ChecksumAddress) -> str:
         return address_to_reverse_domain(address)
 
     def __init__(
-        self, provider: 'BaseProvider' = cast('BaseProvider', default), addr: ChecksumAddress = None
+        self,
+        provider: 'BaseProvider' = cast('BaseProvider', default),
+        middlewares: Optional[Sequence[Tuple['Middleware', str]]] = None,
```

Comment:
This is a breaking change if someone was passing the address as the second param without the keyword before. Is it important for middlewares to be the second arg? Sorry I wasn't there for the review yesterday!